src/Sulu/Bundle/MediaBundle/Api/Collection.php

```diff
@@ -60,19 +115,48 @@ public function __construct(CollectionInterface $collection, $locale)
     }
 
     /**
-     * @VirtualProperty
-     * @SerializedName("children")
-     * @return array
+     * Set children of resource
+     * @param Collection[] $children
+     */
+    public function setChildren($children)
+    {
+        $childrenEntities = array();
+
+        foreach ($children as $child) {
+            $childrenEntities[] = $child->getEntity();
+        }
+
+        // FIXME destroys entity if depth is not deep enough
+        // $this->entity->setChildren(new ArrayCollection($childrenEntities));
+
+        // FIXME cache for children because of the preview images
+        $this->children = $children;
+    }
+
+    /**
+     * Returns children of resource
+     * @return Collection[]
      */
     public function getChildren()
     {
-        $childIds = array();
-        /** @var Entity $child */
-        foreach ($this->entity->getChildren() as $child) {
-            $childIds[] = $child->getId();
+        // FIXME remove cache for children because of the preview images
```

Comment:
Should the children be cached or not? Somehow conflicts with the comment in the previous method for me.